bin/main.ml

```diff
@@ -130,6 +133,43 @@ module Log = struct
     Log.create ~display:common.config.display ()
 end
 
+let watch_command root_path =
+  let excludes = [ {|\.#|}
+                 ; {|_build|}
+                 ; {|\.hg|}
+                 ; {|\.git|}
+                 ; {|~$|}
+                 ; {|/#[^#]*#$|}
+                 ; {|\.install$|}
+                 ]
+  in
+  let path = Path.to_string_maybe_quoted root_path in
+  match Bin.which "inotifywait" with
+  | Some inotifywait ->
+    (* On Linux, use inotifywait. *)
+    let excludes = String.concat ~sep:"|" excludes in
+    inotifywait, ["-r"; path; "--exclude"; excludes; "-e"; "close_write"; "-q"]
+  | None ->
+    (* On all other platforms, try to use fswatch. fswatch's event
+       filtering is not reliable (at least on Linux), so don't try to
+       use it, instead act on all events. *)
+    (match Bin.which "fswatch" with
+     | Some fswatch ->
+       let excludes =
+         List.(map ~f:(fun x -> ["--exclude"; x]) excludes |> concat)
+       in
+       fswatch, ["-r"; path; "-1"] @ excludes
+     | None ->
+      (* Exit immediately to prevent a loop of these errors. *)
+      Format.eprintf "Error: fswatch not found. \
```

Comment:
To match the style of other errors, could you use:

```ocaml
die "@{<error>Error@}: fswatch ..."
```

The message needs to mention inotify as well since we look for either of these tools